src/OrchardCore.Modules/OrchardCore.Email/Services/SmtpSettingsConfiguration.cs

```diff
@@ -40,11 +49,19 @@ public void Configure(SmtpSettings options)
             options.UseDefaultCredentials = settings.UseDefaultCredentials;
             options.UserName = settings.UserName;
 
-            // Decrypt the password
-            if (!String.IsNullOrWhiteSpace(settings.Password))
+            if (!String.IsNullOrEmpty(settings.PasswordSecret))
+            {
+                var shellScope = _shellHost.GetScopeAsync(_shellSettings).GetAwaiter().GetResult();
+                shellScope.UsingAsync(async scope => {
+                    var textSecretService = scope.ServiceProvider.GetRequiredService<ISecretService<TextSecret>>();
+                    options.Password = (await textSecretService.GetSecretAsync(settings.PasswordSecret)).Text;
+                }).GetAwaiter().GetResult();
```

Comment:
@jtkech I forgot but wanted to check with you, particularly as you are now thinking that the `ISiteService` should really be scoped.

Is this the correct way to resolve a scoped service through a singleton?

Reply:
Yes, except of the many `.GetAwaiter().GetResult()` but only on startup.

As it is for creating a child scope, you can also use `ShellScope.CreateChildScopeAsync()` which does the same

        /// <summary>
        /// Creates a child scope from the current one.
        /// </summary>
        public static Task<ShellScope> CreateChildScopeAsync()
        {
            var shellHost = ShellScope.Services.GetRequiredService<IShellHost>();
            return shellHost.GetScopeAsync(ShellScope.Context.Settings);
        }

And then maybe the lighter `UsingServiceScopeAsync()` to not use the activating / terminating handlers. Normally as a child scope, terminating handlers can't be called, but here in an `IConfiguration<>` the shell may not be activated yet.

But that's okay to use `.UseAsync()` as long as the related options are not resolved through a service used while activating the shell. Look at `OpenIdValidationConfiguration` this is what it does

Otherwise you could have just used

    await ShellScope.UsingChildScopeAsync()

Here same kind of problem as in #7953, need to focus on them again step by step ;)

For now i would suggest to do what you are doing, and also use the new `activateShell` parameter set to false.

                shellScope.UsingAsync(async scope =>
                {
                    var textSecretService = scope.ServiceProvider.GetRequiredService<ISecretService<TextSecret>>();
                    options.Password = (await textSecretService.GetSecretAsync(settings.PasswordSecret)).Text;
                }, activateShell: false).GetAwaiter().GetResult();